### ISSUE TITLE
Index expression not narrowed with negative index

### ISSUE THREAD

**Issue by hauntsaninja** [microsoft/pyright#3150]
**Describe the bug**

```
def f(x: list[str | int]) -> None:
    if isinstance(x[0], str):
        reveal_type(x[0])  # reveals str
    if isinstance(x[-1], str):
        reveal_type(x[-1])  # reveals str | int
```

**Expected behavior**

I'd expect the type checker to narrow in both cases. For what it's worth, mypy does narrow in both cases.

I guess that maybe there's an argument to be had that appending to a list is much more common than prepending or deleting and so narrowing is here is slightly less safe? But I don't know that I would really buy such an argument.

**Comment by erictraut** [comment]
Yeah, when I initially added support for narrowing of index expressions, I was worried about supporting negative subscripts because of the possibility of appending. But if mypy supports this, it's probably fine.

I've added support for negative literal integer subscripts for all type narrowing patterns. This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.228, which I just published. It will also be included in the next release of pylance.
